src/Cache.jl

```diff
@@ -0,0 +1,95 @@
+"""
+    Cache
+
+The `Cache` module implements a cache to be used by other modules.
+
+Currently, we have implemented a Least Recently Used (LRU) cache. The cache is
+implemented as a dictionary with a maximum size. When the cache is full, the
+least recently used item is removed to make space for the new item.
+
+The cache is used by the `FileReaders` module to store files that are currently
+open, and by the `DataHandler` module to store regridded fields.
+"""
+
+module Cache
+
+import Base: Callable
+
+export LRUCache, get!, enforce_size!
+
+struct LRUCache{K, V}
+    cache::Dict{K, V}
+    max_size::Int
+    priority::Vector{K}
+end
+
+"""
+    LRUCache{K, V}(max_size::Int) where {K, V}
+
+Construct an empty `LRUCache` with a maximum size of `max_size`.
+"""
+function LRUCache{K, V}(max_size::Int) where {K, V}
+    return LRUCache{K, V}(Dict{K, V}(), max_size, Vector{K}())
+end
+
+"""
+    Base.get!(cache::LRUCache{K, V}, key::K, default::V) where {K, V}
+
+Get the value associated with `key` in `cache`. If the key is not in the
+cache, add it with the value `default`. In any case, update the key's priority
+and make sure the cache doesn't exceed its maximum size.
+"""
+function Base.get!(cache::LRUCache{K, V}, key::K, default::V) where {K, V}
+    if haskey(cache.cache, key)
+        # Remove key from priority queue so we can update it
+        filter!(k -> k != key, cache.priority)
+    else
+        cache.cache[key] = default
+    end
+    # Update priority of key
+    push!(cache.priority, key)
+    enforce_size!(cache)
+    return cache.cache[key]
```

Comment:
Maybe something like this:
```suggestion
update_priority!(cache, key)
enforce_size!(cache)
return get!(cache.cache, key, default)
```
With 
```julia
function update_priority(cache, key)
filter!(k -> k != key, cache.priority)
push!(cache.priority, key)
```
(In either cases we want the new key to be the last entry). 

And `enforce_size` that enforces `max_size - 1`

Reply:
`cache.cache` is a Dictionary so this is not recursive. It is implementing our `get!` in terms of Julia's `get!`, which

Reply:
Ahh, I missed that it was calling `get!` on `cache.cache`